devtools/x/src/nextest.rs

```diff
@@ -36,17 +45,21 @@ pub struct Args {
     #[structopt(long)]
     /// Do not run tests, only compile the test executables
     no_run: bool,
+
     /// Run ignored tests
     #[structopt(long, possible_values = &RunIgnored::variants(), default_value, case_insensitive = true)]
     run_ignored: RunIgnored,
+    /// Test partition, e.g. hash:1/2 or count:2/3
+    #[structopt(long)]
+    pub partition: Option<BuildPartitioner>,
```

Comment:
unless we're partitioning xbuild and others as well, should this be TestPartitioner?   The build of the necessary binaries is incidental, no.

Reply:
Ahh so `BuildPartitioner` is called that because its API is modeled after [`BuildHasher`](https://doc.rust-lang.org/std/hash/trait.BuildHasher.html). The reason for that is that an arbitrary partitioner, in order to split tests, may need to maintain state internally. (A hash-based partitioner might not but a counting-based partitioner has to.) So a `BuildPartitioner` can be used to create ("build") a per-test-binary partitioner, which can have mutable internal state.

I guess a more Java-y name for this would be `TestPartitionerFactory`, and given the confusion around the multiple meanings of "build" I guess I'll change this to that.

Reply:
PartitionBuilder would be clearer, not everything in java is a Factory.... ;).  Besides Factories tend to appear to be stateless. 🤷

Reply:
Yeah I'll call it `TestPartitionerBuilder`.